pandas/core/index.py

```diff
@@ -5004,8 +5004,13 @@ def drop(self, labels, level=None, errors='raise'):
                 loc = self.get_loc(label)
                 if isinstance(loc, int):
                     inds.append(loc)
-                else:
+                elif isinstance(loc, slice):
                     inds.extend(lrange(loc.start, loc.stop))
+                elif is_bool_indexer(loc):
```

Comment:
You're right. 

I'll start an issue and a PR for the KeyError (bug 2 mentioned in the first message in this PR)